backends/bevy_picking_sprite/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+//! A raycasting backend for [`bevy_sprite`](bevy::sprite).
+
+#![allow(clippy::type_complexity)]
+#![allow(clippy::too_many_arguments)]
+#![deny(missing_docs)]
+
+use std::cmp::Ordering;
+
+use bevy::ui::FocusPolicy;
+use bevy::{prelude::*, window::PrimaryWindow};
+use bevy_picking_core::backend::prelude::*;
+
+/// Commonly used imports for the [`bevy_picking_sprite`](crate) crate.
+pub mod prelude {
+    pub use crate::SpriteBackend;
+}
+
+/// Adds picking support for [`bevy_sprite`](bevy::sprite)
+#[derive(Clone)]
+pub struct SpriteBackend;
+impl PickingBackend for SpriteBackend {}
+impl Plugin for SpriteBackend {
+    fn build(&self, app: &mut App) {
+        app.add_system(sprite_picking.in_set(PickSet::Backend));
+    }
+}
+
+/// Checks if any sprite entities are under each pointer
+pub fn sprite_picking(
+    pointers: Query<(&PointerId, &PointerLocation)>,
+    cameras: Query<(Entity, &Camera)>,
+    primary_window: Query<Entity, With<PrimaryWindow>>,
+    images: Res<Assets<Image>>,
+    sprite_query: Query<(
+        Entity,
+        &Sprite,
+        &Handle<Image>,
+        &GlobalTransform,
+        &ComputedVisibility,
+        Option<&FocusPolicy>,
+    )>,
+    mut output: EventWriter<EntitiesUnderPointer>,
+) {
+    let mut sorted_sprites: Vec<_> = sprite_query.iter().collect();
+    sorted_sprites.sort_by(|a, b| {
+        (b.3.translation().z)
+            .partial_cmp(&a.3.translation().z)
+            .unwrap_or(Ordering::Equal)
+    });
+
+    for (pointer, location) in pointers.iter().filter_map(|(pointer, pointer_location)| {
+        pointer_location.location().map(|loc| (pointer, loc))
+    }) {
+        let mut blocked = false;
+        let (cam_entity, camera) = cameras
+            .iter()
+            .find(|(_entity, camera)| {
+                camera
+                    .target
+                    .normalize(Some(primary_window.single()))
+                    .unwrap()
+                    == location.target
+            })
+            .unwrap_or_else(|| panic!("No camera found associated with pointer {:?}.", pointer));
+        let target_half_extents = if let Some(target) = camera.logical_target_size() {
+            target / 2.0
+        } else {
+            continue;
+        };
+        let cursor_position = location.position;
+        let cursor_pos_centered = cursor_position - target_half_extents;
+
+        let picks: Vec<(Entity, PickData)> = sorted_sprites
+            .iter()
+            .copied()
```

Comment:
I'm not sure this copy is still needed ? As the filter_map will create new instances not referring to original data

Reply:
`iter()` yields references. The `copied` is needed to return owned data. The alternative is to dereference in the code that follows, but it will be copying regardless, which adds noise for no benefit. The `copied` makes the `Entity` and `Option<&FocusPolicy>` owned instead of references.